src/corehost/src/args.cpp

```diff
@@ -43,6 +45,15 @@ bool parse_arguments(const int argc, const pal::char_t* argv[], arguments_t& arg
             return false;
         }
         args.managed_application = pal::string_t(argv[1]);
+
+        // Take care of app_dir
+        args.app_dir = args.managed_application;
+        if (!pal::realpath(args.app_dir))
```

Comment:
`managed_application` should probably be `realpath`-ized as well. Then you could assume `app_dir` starts as an absolute path so you'd still only be hitting the disk once.